### ISSUE TITLE
MathCAT does not support portable installation (e.g. via scoop)

### ISSUE THREAD

**Issue by xworld21** [NSoiffer/MathCATForPython#79]
MathCAT uses a hardcoded path for the location of NVDA: it has to be in `\\AppData\\Roaming\\nvda`. However, when installed via scoop, all the data is in `\\scoop\\apps\\nvda\\current\\userConfig`, and MathCAT fails right away.

Would it be possible to detect the correct folder dynamically?

**Comment by NSoiffer** [comment]
Sorry for the slow response...

MathCAT sets the Rules dir with
```
rules_dir = path.join(path.dirname(path.abspath(__file__)), "Rules")
```
It is looking up its current location and adding the subdir "Rules". There is no hardcoded path. Any idea why this doesn't work for a portable installation?

**Comment by xworld21** [comment]
I was referring to lines like
https://github.com/NSoiffer/MathCATForPython/blob/5d4b179aa3e4d7b61ecc199e02db9a612b60d799/addon/globalPlugins/MathCAT/MathCATPreferences.py#L51
I tried to fix `MathCATPreferences.py` in my local install, and that made MathCAT at least launch without crashing, but I still couldn't read MathML from Chrome, strangely enough.

**Comment by NSoiffer** [comment]
Thanks for pointing that out. Someone else helped out and wrote that code. I'll change it to use `path.dirname(path.abspath(__file__))`.

Will the user preference being stored in `~/AppData/Roaming/MathCAT` cause a problem?

**Comment by xworld21** [comment]
Thanks! Yes MathCAT is not crashing anymore and I can open the settings. It still reads MathML in Chrome as 'math', but I am not sure if it's my own misconfiguration.

For people using properly portable NVDA (as in NVDA on a USB stick), than the MathCAT settings should try to be in the same place as NVDA. (Do people use NVDA in this way? I don't know! It's fine for me that the issue stays closed until somebody actually hits this variant of the problem.)

> Will the user preference being stored in `~/AppData/Roaming/MathCAT` cause a problem?

While we are discussing this, I'd recommend using `%APPDATA%` with... I believe `os.path.expandvars()`, instead of hardcoding `AppData/Roaming`.

**Comment by NSoiffer** [comment]
Thanks for the suggestion. I switched to using `%APPDATA%`.
